rest_framework_jwt/serializers.py

```diff
@@ -57,10 +57,25 @@ def validate(self, attrs):
 
                 payload = jwt_payload_handler(user)
 
-                return {
+                data = {
                     'token': jwt_encode_handler(payload),
                     'user': user
                 }
+                if api_settings.JWT_PERMANENT_TOKEN_AUTH:
+                    headers = self.context['request'].META
+                    device_name = headers.get('HTTP_X_DEVICE_MODEL')
+                    user_agent = headers.get('HTTP_USER_AGENT')
+                    if not device_name:
+                        device_name = user_agent
+                        device_details = ''
+                    else:
+                        device_details = user_agent
+
+                    data['permanent_token'] = Device.create(
+                        user=user, jwt_secret=data['token'], last_request_datetime=datetime.now(),
```

Comment:
what? JWT secred is not 
data = {
                      'token': jwt_encode_handler(payload),		                      'token': jwt_encode_handler(payload)..

JWT_secret should be generated as UUID.